Sources/AppcuesKit/Presentation/UI/ExperienceStepViewModel.swift

```diff
@@ -20,6 +20,12 @@ internal class ExperienceStepViewModel: ObservableObject {
     private let actions: [UUID: [Experience.Action]]
     private let actionRegistry: ActionRegistry?
 
+    @Published private var formComponents: [UUID: FormItem]
+
+    var formIsComplete: Bool {
+        !formComponents.contains { !$0.value.isSatisfied }
+    }
```

Comment:
assume this is coming later, but curious where this ties in to the action flow

Reply:
So I was experimenting with something in the Button component that checks `viewModel.formIsComplete` and disables the button, but that won't work if we only want to disable some buttons (depending on the action).

Reply:
I think only the continue action. The tricky part is if we want the button to look disabled? I think web flows don't disable the button (and in general that's a better UX). If we don't disable the button, then in theory the continue action (or the ExperienceRenderer method that it calls) needs to somehow check the state in the view model and not do anything if it's invalid (and maybe show an inline error message or indicator that the field is required?)

Reply:
First thought is perhaps some optional validation function on ExperienceAction's that would return valid by default but allow an override hook to let any action apply validation - in a hopefully non-breaking way to the public API 🤔 .  Would also need to figure out how to package up the right set of params to that function to be able to validate against.